owm2ffmap/owm2ffmap.py

```diff
@@ -36,6 +36,30 @@
 bounding_box_elems = [float(x) for x in bounding_box.split(",")]
 
 
+
+def hash_firmware(firmware: str) -> int:
+    """This function hashes a firmware string into an int to make
+    versions compareable. early versions get smaller numbers, more
+    recent ones get bigger numbers.
+    Frimware-Strings that do not follow the semantic versioning scheme
+    are considered pre-kathleen (return -1).
+
+    Args:
+        firmware (str): Name-string of a firmware with numbers at the end
+
+    Returns:
+        int: hash-value. If firmware doesn't follow semantiv-versioning
+        scheme, return -1.
+    """
+    # firmware-string matches semantic-versioning
+    version = re.search(".*([0-9]\.[0-9]\.[0-9]*)", firmware)
+    if version:
+        return int(version.group(1).replace('.',''))
```

Comment:
Think this will only work for single-digit numbers in minor and patch. comparing v1.0.10 and v1.1.0 should end up in `1010 > 110`)

Reply:
That is right. By current development tempo, I silently assumed, that we won't have more than 9 point releases per version in the next time. That would complicate things more.

Feel free to provide a patch, if you consider this issue important. :)

Reply:
added a comment in 98a1e4910fa01aaa7f784bd2b7d86217f58386f5

Reply:
0ff0a93 should be shorter, correct, and better readable. (just a side note, the previous function computed some version _ordinal_ number rather than a hash)

I also added regression tests. Run the script with `--update-tests` and it will write inputs and outputs to `testdata/`. Run it again with `--tests` and it will verify the results are still the same.